packages/astro/test/fixtures/astro-dynamic/src/pages/client-only.astro

```diff
@@ -0,0 +1,11 @@
+---
+import Counter from '../components/Counter.jsx';
+import SvelteCounter from '../components/SvelteCounter.svelte';
```

Comment:
These imports should be removed so that the code doesn't run in the server.

Reply:
How would Astro know which components to reference on the client? I haven’t read through all of this, but if it’s by file or export name that’s guaranteed to cause conflicts and confusion.

Reply:
maybe `import "./Component.jsx?client";` would work?

Reply:
There’s also precedent in the form of `lazy(() => import(…))`, which could be `client(() => import(…))` in this case, and doesn’t require further fracturing import statements just as they’ve begun to be statically analyzable in a ~predictable way.

Reply:
And that would actually be pretty easy to write as a function right? `client()` is just a predicate that calls the function called to it if a set of conditions pass?

Reply:
Sorry for the confusion, the imports should exist in the source code and be removed by the compiler when turned into JavaScript.

Reply:
@matthewp’s answer is better than any workaround I might suggest.